lib/src/list_multimap/list_multimap_builder.dart

```diff
@@ -132,22 +132,20 @@ class ListMultimapBuilder<K, V> {
     });
   }
 
-  /// As [ListMultimap.remove] but returns nothing.
-  void remove(Object key, V value) {
-    if (key is K) {
-      _makeWriteableCopy();
-      _getValuesBuilder(key).remove(value);
-    }
+  /// As [ListMultimap.remove].
+  bool remove(Object key, V value) {
+    if (key is! K) return false;
+    _makeWriteableCopy();
+    return _getValuesBuilder(key).remove(value);
   }
 
-  /// As [ListMultimap.removeAll] but returns nothing.
-  void removeAll(Object key) {
-    if (key is K) {
-      _makeWriteableCopy();
-
-      _builtMap = _builtMap;
-      _builderMap[key] = new ListBuilder<V>();
-    }
+  /// As [ListMultimap.removeAll], but results are [ListBuilder]s.
+  ListBuilder<V> removeAll(Object key) {
+    if (key is! K) return new ListBuilder<V>();
+    _makeWriteableCopy();
+    final old = _getValuesBuilder(key);
+    _builderMap[key] = new ListBuilder<V>();
+    return old;
```

Comment:
Returning a `ListBuilder` seems misleading; it's no longer associated with anything, so any modifications you make to it will be lost.

It'll hurt performance a bit, but I think returning a `BuiltList` would a better. What do you think?

Reply:
Yeah, that sounds good. I wasn't sure between `ListBuilder` and `BuiltList` here.

It should be possible to do:
```suggestion
    _makeWriteableCopy();
    final old = _builderMap[key]?.build() ?? _builtMap[key] ?? new BuiltList<V>();
    _builderMap[key] = new ListBuilder<V>();
    return old;
```
which I think should have better performance because it'll normally return the existing BuiltList.

Reply:
(Sorry for slow response, have been sick this week!)

Yes looks reasonable.

Reply:
(No worries, hope you're feeling better!)